File: BBB.Framework.Dto/Core/RaidDto.cs
namespace BBB.Framework.Dto.Core
{
    public sealed class RaidDto
    {
        public Guid RaidKey { get; set; }

        public string Name { get; set; } = string.Empty;

        public List<EncounterDto> Encounters { get; set; } = new();
    }
}

File: BBBRaiderWeb/src/models/RaidDto.ts
import { logTypeMissmatch, parseArray, parseString } from '../helpers/model'
import { EncounterDto } from './EncounterDto'

export class RaidDto {
  raidKey: string = ''
  name: string = ''
  encounters: EncounterDto[] = []

  static fromJSON(obj: any): RaidDto {
    if (!obj) {
      logTypeMissmatch('RaidDto', obj)
      return new RaidDto()
    }

    return {
      raidKey: parseString(obj.raidKey),
      name: parseString(obj.name),
      encounters: parseArray(obj.encounters, EncounterDto.fromJSON),
    }
  }
}
